src/Jackalope/Item.php

```diff
@@ -636,10 +636,6 @@ protected function checkState()
             $this->postDirtyState = -1;
         }
 
-        if ($this->state === self::STATE_DELETED) {
-            throw new InvalidItemStateException('Item '.$this->path.' is deleted');
-        }
```

Comment:
the problem with this is that when you manipulate the node, it calls checkState to make sure it can be manipulated. 

looking at the stack trace in https://travis-ci.org/jackalope/jackalope-doctrine-dbal/jobs/119277489 i think the problem is something else. https://github.com/jackalope/jackalope/blob/check_state_delete_no_exception/src/Jackalope/ObjectManager.php#L1124 assumes that a node that is dirty is not deleted (because both are states). but then ->getIdentifier will trigger checkState which detects the node is deleted. i think before the deleted check in the object manager, we should check for dirty and if the node is dirty call refresh($keepChanges). then the state will not be dirty anymore but deleted or new or modified and things work as expected i'd hope.